File: Solutions/0303- Range Sum Query - Immutable/Solution.py
class NumArray:
    my_array = []

    def __init__(self, nums: List[int]):
        self.my_array = nums
        

    def sumRange(self, left: int, right: int) -> int:
        sum = 0
        while left <= right:
            sum +=  self.my_array[left]
            left += 1
        return sum

File: Solutions/0303- Range Sum Query - Immutable/Solution.java
class NumArray {
    private int[] myArray;

    public NumArray(int[] nums) {
        int size = nums.length;
        myArray = new int[size + 1];
        for (int i = 0; i < size; i++) {
            myArray[i+1] = myArray[i] + nums[i];
        }
    }

    public int sumRange(int left, int right) {
        return myArray[right + 1] - myArray[left];
    }
}